clusterloader2/pkg/measurement/common/scheduler_latency.go

```diff
@@ -134,37 +146,110 @@ func (*schedulerLatencyMeasurement) String() string {
 	return schedulerLatencyMetricName
 }
 
-func (s *schedulerLatencyMeasurement) resetSchedulerMetrics(c clientset.Interface, host, provider, masterName string, masterRegistered bool) error {
-	_, err := s.sendRequestToScheduler(c, "DELETE", host, provider, masterName, masterRegistered)
-	if err != nil {
-		return err
+// Helper function to substract two histograms
+func histogramSub(finalHist, initialHist *measurementutil.Histogram) *measurementutil.Histogram {
+	for k := range finalHist.Buckets {
+		finalHist.Buckets[k] = finalHist.Buckets[k] - initialHist.Buckets[k]
 	}
-	return nil
+	return finalHist
 }
 
-// Retrieves scheduler latency metrics.
-func (s *schedulerLatencyMeasurement) getSchedulingLatency(c clientset.Interface, host, provider, masterName string, masterRegistered bool) ([]measurement.Summary, error) {
+func (m *schedulerLatencyMetrics) substract(sub schedulerLatencyMetrics) {
+	if sub.preemptionEvaluationHist != nil {
+		m.preemptionEvaluationHist = histogramSub(m.preemptionEvaluationHist, sub.preemptionEvaluationHist)
+	}
+	if sub.schedulingAlgorithmDurationHist != nil {
+		m.schedulingAlgorithmDurationHist = histogramSub(m.schedulingAlgorithmDurationHist, sub.schedulingAlgorithmDurationHist)
+	}
+	if sub.e2eSchedulingDurationHist != nil {
+		m.e2eSchedulingDurationHist = histogramSub(m.e2eSchedulingDurationHist, sub.e2eSchedulingDurationHist)
+	}
+	for _, ep := range extentionsPoints {
+		if sub.frameworkExtensionPointDurationHist[ep] != nil {
+			m.frameworkExtensionPointDurationHist[ep] = histogramSub(m.frameworkExtensionPointDurationHist[ep], sub.frameworkExtensionPointDurationHist[ep])
+		}
+	}
+}
+
+func (s *schedulerLatencyMeasurement) setQuantiles(metrics schedulerLatencyMetrics) (schedulingMetrics, error) {
 	result := schedulingMetrics{
 		FrameworkExtensionPointDuration: make(map[string]*measurementutil.LatencyMetric),
 	}
-	data, err := s.sendRequestToScheduler(c, "GET", host, provider, masterName, masterRegistered)
+	for _, ePoint := range extentionsPoints {
+		result.FrameworkExtensionPointDuration[ePoint] = &measurementutil.LatencyMetric{}
+	}
+
+	if err := s.setQuantileFromHistogram(&result.E2eSchedulingLatency, metrics.e2eSchedulingDurationHist); err != nil {
+		return result, err
+	}
+	if err := s.setQuantileFromHistogram(&result.SchedulingLatency, metrics.schedulingAlgorithmDurationHist); err != nil {
+		return result, err
+	}
+
+	for _, ePoint := range extentionsPoints {
+		if err := s.setQuantileFromHistogram(result.FrameworkExtensionPointDuration[ePoint], metrics.frameworkExtensionPointDurationHist[ePoint]); err != nil {
+			return result, err
+		}
+	}
+
+	if err := s.setQuantileFromHistogram(&result.PreemptionEvaluationLatency, metrics.preemptionEvaluationHist); err != nil {
+		return result, err
+	}
+	return result, nil
+}
+
+// Retrieves scheduler latency metrics.
```

Comment:
Same here and below

// getSchedulingLatency retrieves....

Reply:
fixed